pkg/descheduler/strategies/duplicates.go

```diff
@@ -46,12 +55,43 @@ func RemoveDuplicatePods(ds *options.DeschedulerServer, strategy api.Descheduler
 // deleteDuplicatePods evicts the pod from node and returns the count of evicted pods.
 func deleteDuplicatePods(client clientset.Interface, policyGroupVersion string, nodes []*v1.Node, dryRun bool, nodepodCount utils.NodePodEvictedCount, maxPodsToEvict int, evictLocalStoragePods bool) int {
 	podsEvicted := 0
+	cepm := creatorEvictedPods{}
 	for _, node := range nodes {
 		klog.V(1).Infof("Processing node: %#v", node.Name)
 		dpm := ListDuplicatePodsOnANode(client, node, evictLocalStoragePods)
 		for creator, pods := range dpm {
+			if _, found := cepm[creator]; !found {
+				cepm[creator] = 0
+			}
+			status, err := getStatusForOwnerRef(client, creator)
+			if err != nil {
+				klog.Errorf("Error getting the number of replicas: %v", err)
+				continue
+			}
+			// bestRatio := float64(status.replicas) / float64(len(nodes))
+			bestRatio := (int(status.replicas) / len(nodes)) + 1
+			// here we have to implement the logic to skip eviction
+			// 1. too few available replicas
+			klog.V(3).Infof("%#v with only %.2f%% of available pods, skipping?", creator, (float64(status.available)/float64(status.replicas))*100.0)
+			if (float64(status.available) / float64(status.replicas)) < 0.6 {
+				klog.V(1).Infof("%#v with only %.2f%% of available pods, skipping", creator, (float64(status.available)/float64(status.replicas))*100.0)
+				continue
+			}
+			// 2. in case the replicas is greater than the number of nodes and we are enough close to the best balance
+			klog.V(3).Infof("%#v on this node has close enough to the best ratio (%d) number of pods %d over replicas %d, skipping?", creator, bestRatio, len(pods), status.replicas)
+			if len(pods) <= (bestRatio + 1) {
```

Comment:
`len(pods)` (in this context) gives a number of pods with the same owner for a node. So you are testing if the number of pods with the same owner is less than the uniform ratio. If so, you skip since you want to balance the pod distribution. Why `bestRatio + 1` then? Is not `bestRatio` sufficient?